spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -303,66 +305,128 @@ public String toString() {
         }
       }
     }
+
+    private class EncryptedOutputFileFactory implements OutputFileFactory<EncryptedOutputFile> {
+      private final int partitionId;
+      private final long taskId;
+      private final long epochId;
+
+      EncryptedOutputFileFactory(int partitionId, long taskId, long epochId) {
+        this.partitionId = partitionId;
+        this.taskId = taskId;
+        this.epochId = epochId;
+      }
+
+      private String generateFilename() {
+        return format.addExtension(String.format("%05d-%d-%s", partitionId, taskId, UUID.randomUUID().toString()));
```

Comment:
The UUID should be the write's UUID so that all of the files for a write can be located by path. The purpose of this is to be able to know from two paths that they were written by the same operation. That's useful, for example, if a Spark job dies and leaves files in the file system, you can identify them all with a recursive listing and grep.

Reply:
Using the same UUID would make filenames collide, so we should have a counter as well?

Reply:
I was under the impression the UUID was just for filename uniqueness. Will fix, and add your comment on the code so that its clear what the purpose is.

Reply:
Right, it will require a counter now.

Reply:
Fixed.